include/materials/mohr_coulomb.tcc

```diff
@@ -409,22 +414,22 @@ Eigen::Matrix<double, 6, 1> mpm::MohrCoulomb<Tdim>::compute_stress(
   // Update MC parameters using a linear softening rule
   if (softening_ && pdstrain > pdstrain_peak_) {
     if (pdstrain < pdstrain_residual_) {
-      (*state_vars)["phi"] =
+      (*state_vars).at("phi") =
           phi_residual_ +
           ((phi_peak_ - phi_residual_) * (pdstrain - pdstrain_residual_) /
            (pdstrain_peak_ - pdstrain_residual_));
-      (*state_vars)["psi"] =
+      (*state_vars).at("psi") =
           psi_residual_ +
           ((psi_peak_ - psi_residual_) * (pdstrain - pdstrain_residual_) /
            (pdstrain_peak_ - pdstrain_residual_));
-      (*state_vars)["cohesion"] =
+      (*state_vars).at("cohesion") =
           cohesion_residual_ + ((cohesion_peak_ - cohesion_residual_) *
                                 (pdstrain - pdstrain_residual_) /
                                 (pdstrain_peak_ - pdstrain_residual_));
     } else {
-      (*state_vars)["phi"] = phi_residual_;
-      (*state_vars)["psi"] = psi_residual_;
-      (*state_vars)["cohesion"] = cohesion_residual_;
+      (*state_vars).at("phi") = phi_residual_;
```

Comment:
Do we need these `.at` checks? they would be checked and initialized at the start. Contracts in C++2x standard will remove this idea of throwing an error with `at`, it's considered an irrecoverable error at this point. I'd just update it using `[]` instead of `at`

Reply:
It's a clean up, solved as discussed privately